File: easy/0482_license_key_formatting/solution.py
class Solution:
    def licenseKeyFormatting(self, s: str, k: int) -> str:
        s = list(s)

        license_key = []
        l = 0

        while s:
            c = s.pop()

            if c == "-":
                continue

            if l == k:
                license_key.append("-")
                l = 0

            license_key.append(c)
            l += 1

        return "".join(reversed(license_key)).upper()

File: easy/0482_license_key_formatting/solution.cpp
#include <string>

class Solution {
public:
  std::string licenseKeyFormatting(std::string s, int k) {
    std::string licenseKey;
    int l = 0;

    while (!s.empty()) {
      char ch = s.back();
      s.pop_back();

      if (ch == '-')
        continue;

      if (l == k) {
        licenseKey.push_back('-');
        l = 0;
      }

      licenseKey.push_back(toupper(ch));
      ++l;
    }

    std::reverse(licenseKey.begin(), licenseKey.end());
    return licenseKey;
  }
};
